File: Programming-languages-and-laboratory/Python/Exercises/22_compito_java_20180622/GestioneUniversita.py
from Corso import Corso
from Teoria import Teoria
from Laboratorio import Laboratorio
from Studente import Studente
from Libretto import Libretto

import sys


class GestioneUniversita:

    def main():

        corsi = []
        studenti = []
        cod_to_studente = {}
        cod_to_corso = {}

        try:
            f = open("corsi.txt", "r")
            line = f.readline().strip()

            while line != "":
                tok = line.split()
                codice = int(tok[0])
                tipo = tok[1]
                nome_corso = f.readline().strip()
                nome_cognome_docente = f.readline().strip()
                
                line = f.readline().strip()
                if tipo == "teoria":
                    tok = line.split()
                    codice_aula = int(tok[0])
                    ore_settimanali = int(tok[1])
                    ore_lezione = float(tok[2])

                    t = Teoria(codice, tipo, nome_corso, nome_cognome_docente, codice_aula, ore_settimanali, ore_lezione)
                    corsi.append(t)
                    line = f.readline().strip()
                    cod_to_corso[codice] = t

                else:
                    nome_laboratorio = line
                    nome_assistente = f.readline().strip()
                    numero_postazioni = int(f.readline().strip())

                    l = Laboratorio(codice, tipo, nome_corso, nome_cognome_docente, nome_laboratorio, nome_assistente, numero_postazioni)
                    corsi.append(l)

                    line = f.readline().strip()
                    cod_to_corso[codice] = l

                line = f.readline().strip()

        except IOError as i:
            print(i)
        except Exception as e:
            print(e)

        print("nome, codice, docente, tipo, aula, ore sett., ore/lez., lab., assistente, postazioni")

        for c in corsi:
            print(c)
        print()

        try:
            f = open("studenti.txt", "r")
            line = f.readline().strip()

            while line != "":
                matricola = int(line)
                nome_cognome_studente = f.readline().strip()
                st = Studente(matricola, nome_cognome_studente)
                studenti.append(st)
                cod_to_studente[matricola] = st

                line = f.readline().strip()

                while line != "":
                    tok = line.split()
                    codice_corso = int(tok[0])
                    voto = tok[1]

                    lib = Libretto(codice_corso, voto)
                    st.add_libretto(lib)
                    st.add_voto(voto, cod_to_corso[codice_corso].get_nome_corso())

                    line = f.readline().strip()
                
                line = f.readline().strip()

        except IOError as i:
            print(i)
        except Exception as e:
            print(e)

        for s in studenti:
            print(s)
        print()

        try:
            codice_letto = int(sys.argv[1])

            for st in studenti:
                if codice_letto == st.get_matricola():
                    print(st.get_nome_studente() + " " + st.get_nome_corso_max() + " " + st.get_voto_max())

        except IOError as i:
            print(i)
        except Exception as e:
            print(e)


if __name__ == "__main__":
    GestioneUniversita.main()

File: 2-second-year-2022-2023/Programming-languages-and-laboratory/Java/Exercises/22_compito_java_2018_06_22/GestioneUniversita.java
import java.util.*;
import java.io.*;

public class GestioneUniversita {
	static List<Corso> corsi = new LinkedList<Corso>();
	static List<Studente> studenti = new LinkedList<Studente>();
	static Map<Integer, String> nomeCorsi = new HashMap<Integer, String>();

	public static void main(String[] args) {

		// punto 1, memorizzazione corsi

		try {
			BufferedReader br = new BufferedReader(new FileReader("corsi.txt"));
			String line = br.readLine();

			while (line != null) {
				StringTokenizer tok = new StringTokenizer(line);
				int codiceCorso = Integer.parseInt(tok.nextToken());
				String tipoCorso = tok.nextToken();

				String nomeCorso = br.readLine();

				String nomeCognomeDocente = br.readLine();

				line = br.readLine();

				if (tipoCorso.equals("teoria")) {

					tok = new StringTokenizer(line);
					int codiceAula = Integer.parseInt(tok.nextToken());
					int oreSettimanali = Integer.parseInt(tok.nextToken());
					double oreLezione = Double.parseDouble(tok.nextToken());

					Corso c = new Teoria(codiceCorso, tipoCorso, nomeCorso, nomeCognomeDocente, codiceAula,
							oreSettimanali, oreLezione);
					corsi.add(c);

					line = br.readLine();

				} else {

					String nomeLaboratorio = line;
					String nomeAssistente = br.readLine();
					int numeroPostazioniLaboratorio = Integer.parseInt(br.readLine());

					Corso c = new Laboratorio(codiceCorso, tipoCorso, nomeCorso, nomeCognomeDocente, nomeLaboratorio,
							nomeAssistente, numeroPostazioniLaboratorio);
					corsi.add(c);

					line = br.readLine();

				}

				line = br.readLine();

			}

			br.close();

		} catch (IOException e) {
			System.err.println(e);
		} catch (Exception e) {
			System.err.println(e);
		}

		// punto 2, memorizzazione studenti

		try {
			BufferedReader br = new BufferedReader(new FileReader("studenti.txt"));
			String line = br.readLine();

			while (line != null) {
				int matricolaStudente = Integer.parseInt(line);
				String nomeCognomeStudente = br.readLine();

				Studente s = new Studente(matricolaStudente, nomeCognomeStudente);
				studenti.add(s);

				line = br.readLine();

				while (line != null && !line.isEmpty()) {
					StringTokenizer tok = new StringTokenizer(line);
					int codiceCorso = Integer.parseInt(tok.nextToken());
					String voto = tok.nextToken();

					Libretto l = new Libretto(codiceCorso, voto);
					s.corsiLibretto.add(l);

					line = br.readLine();

				}

				line = br.readLine();

			}

			br.close();

		} catch (IOException e) {
			System.err.println(e);
		} catch (Exception e) {
			System.err.println(e);
		}
		// punto 3, stampa elenco di tutti i corsi

		System.out.println("nome\tcodice\tdocente\ttipo\taula\tore sett.\tore/lez.\tlab.\tassistente\tpostazioni");

		for (Corso c : corsi) {
			System.out.println(c);
		}

		System.out.println("");

		// punto 4, stampa elenco tutti studenti con nome, media esami e libretto

		for (Studente s : studenti)
			System.out.println(s);
	}
}
